machine.go

```diff
@@ -405,45 +449,70 @@ func (m *Machine) writerKernelModules(w *writerhelper.WriterHelper, moddir strin
 			"modules.symbols",
 			"modules.symbols.bin",
 			"modules.builtin.bin",
-			"modules.devname")
+			"modules.devname"}
 
-	// build a list of built-in modules so that we don’t attempt to copy them
-	var builtinModules = make(map[string]bool)
 
-	f, err := os.Open(path.Join(moddir, "modules.builtin"))
-	if err != nil {
-		return err
+	for _, v := range modfiles {
+		if err := w.CopyFile(moddir + "/" + v); err != nil {
+			return err
+		}
 	}
-	defer f.Close()
 
-	scanner := bufio.NewScanner(f)
-	for scanner.Scan() {
-		module := scanner.Text()
-		builtinModules[module] = true
+	prefix := ""
+	if mergedUsrSystem() {
+		prefix = "/usr"
 	}
 
-	if err := scanner.Err(); err != nil {
-		return err
-	}
+	copiedModules := make(map[string]bool)
 
 	for _, v := range modules {
-		if builtinModules[v] {
+		modname := strings.TrimSuffix(v[strings.LastIndex(v, "/") + 1 :], ".ko")
+		modpath := getModPath(modname)
+
+		if modpath == "" {
+			return errors.New("Modules path couldn't be determined")
+		}
+
+		if modpath == "(builtin)" || copiedModules[modname] {
 			continue
 		}
 
-		modpath := path.Join(moddir, v)
+		if _, err := os.Stat(prefix + modpath); os.IsNotExist(err) {
+			return err
+		}
 
-		if strings.HasSuffix(modpath, ".ko") {
-			if _, err := os.Stat(modpath); err != nil {
-				modpath += ".xz"
+		if err := w.CopyFile(prefix + modpath); err != nil {
+			return err
+		} else {
+			copiedModules[modname] = true
+		}
```

Comment:
Nit: personally I'd drop the explicit `else`, namely:
```
if something
    return foo
 
another()
```
Although if gofmt prefers something else - so be it.